Update OpenShift Pipelines support policy to support less versions on more OpenShift versions

vdemeester opened the pull request:
The plan is to reduce the number of OpenShift Pipelines version we
support at a given time while augmenting the number of OpenShift

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

vdemeester commented on enhancements/support-less-on-more.md:
```diff
@@ -0,0 +1,120 @@
+
+The idea of the proposal is to make it more appealing to get the
+latest release. The only challenge can be the
+`MIN_KUBERNETES_VERSION`, as discussed below.
```
It's upgraded about once or twice per year, so it might happen more or less every 2 releases.

concaf commented:
@vdemeester please merge

concaf commented on enhancements/support-less-on-more.md:
```diff
@@ -0,0 +1,146 @@
+would mean supporting 1.11 and 1.10.
+
+In addition, as soon as release N is released, release N-1 becomes in
+maintenance mode, aka gets only critical bugfixes and security
```
Koustav brought up that we expect customers to deploy N and N-1 in prod and it's not the best scenario to not support N-1 in prod and pushing the upgrade to N all the times; we're already reducing release support footprint to 2 releases, but we need to be decently active on the N-1 release as well

vdemeester replied:
Define "decently active". We can also define what maintenance support means for us btw. We won't push new features for N-1 release, but we will backport bugfix is affected, in theory, we would do bugfix release when upstream LTS also gets bugfix releases, etc…

ppitonak replied:
I personally don't care how we call it officially. What I care is the amount of work we actually do - IMHO we should limit it as much as possible for N-1

concaf commented on enhancements/support-less-on-more.md:
```diff
@@ -0,0 +1,146 @@
+	- Not supported in 2024-12 (OSP 1.18 + 1 month)
+- … and so one
+
+In this example, OSP 1.13 would need to be supported, *in maintenance mode*, until 17th January 2025 as it's the "highest" version that works on 4.12, which is an EUS.
```
i see this as us having to support 3 versions of pipelines still due to EUS constraints, even if it will be maintenance support 🤔

vdemeester replied:
Yes, there will be no way around this. It depends on how quick or slow tekton moves the minimum kubernetes version value though.

concaf replied:
OCP 4.12
- maintenance support ends: July 17, 2024
- EUS support ends: January 17, 2025

we are releasing 1.16 on September 17, 2024, which is "after" maintenance support but "within" EUS support window AND min kube version is satisfied
- we will not release 1.16 on OCP 4.12
- we will continue supporting the latest available release (1.15) on OCP 4.12 till EUS support ends

i.e. for a given EUS release, we support the latest shipped pipelines version that was shipped before maintenance support ended and we will not ship a new one

OR

we are releasing 1.16 on September 17, 2024, which is "after" maintenance support but "within" EUS support window AND min kube version is satisfied
- we will release 1.16 on OCP 4.12
IF ELSE min kube version is not satisfied, then continue support 1.15 on OCP 4.12

---

OCP 4.13
- maintenance support ends: November 17, 2024	

we are releasing 1.17 on December 31, 2024, then:
- we will ship it for OCP 4.12 but we will not ship it for OCP 4.13
- we will ship it for OCP 4.14 onwards (whatever the supported OCP versions are)

concaf commented on enhancements/support-less-on-more.md:
```diff
@@ -0,0 +1,146 @@
+latest release. The only challenge can be the
+`MIN_KUBERNETES_VERSION`, as discussed below.
+
+### Support down to `MIN_KUBERNETES_VERSION`
```
when a new OCP is released and minimum k8s version is satisfied for N-1 version, then do we release N-1 on the new OCP version as well?
for example: 1.11 and 1.12 are supported on 4.12, 4.13, 4.14; now when 4.15 is released, will 1.11 support 4.15 as well?

vdemeester replied:
Yes *if* there is not already 2 releases on that versions. In 4.15 case, we should already have 1.12 and 1.13, so we don't really need to add 4.15 to 1.11

RickJWagner replied:
I think that will work *once* in the given use case.
Say time goes by and we have Pipelines 1.14 GA.
We would not want to de-support Pipelines 1.12 on OCP 4.15 at that time.
(Reason:  Say a customer is on OCP 4.12.  They have available Pipelines 1.11, 1.12.  They choose to upgrade OCP.
- The newest Pipelines they can be on is 1.12, they cannot upgrade to OCP 4.15, which will not support 1.12.  They are 'stuck')

vdemeester replied:
@RickJWagner if 1.12 is not "supported" on 4.15 it would *still* not block the upgrade as, technically it still runs, *and* in the "upgrade path", 1.14 comes from 1.13 which comes from 1.12, so the "chain" shouldn't be broken imo.

concaf replied:
decision: while shipping a new minor release, we take the minimum k8s version in account to add the supported OCP versions for that release - https://access.redhat.com/solutions/4870701

concaf commented on enhancements/support-less-on-more.md:
```diff
@@ -0,0 +1,120 @@
+would mean supporting 1.11 and 1.10.
+
+In addition, as soon as release N is released, release N-1 becomes in
+maintenance mode, aka gets only critical bugfixes and security
```
case in point: the pipelines performance and HA config and fixes we shipped for 1.10, etc <-- this involved features (like new fields, etc) and bugfixes

decision: we should push back when asked for such requests ^^^ also, this proposal will fix this by _design_ as we will support less releases on more OCP (supporting min k8s version)
delta: push customers to upgrade more and be less lenient and less eager with shipping fixes/features in older releases

concaf replied:
--- thread resolved ---

vdemeester committed:
Update OpenShift Pipelines support policy to support less versions on more OpenShift versions

The plan is to reduce the number of OpenShift Pipelines version we
support at a given time while augmenting the number of OpenShift

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

concaf reviewed (approved):
/lgtm
